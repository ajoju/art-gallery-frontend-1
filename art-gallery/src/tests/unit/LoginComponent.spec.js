import { mount } from "@vue/test-utils";
import LoginComponent from "../../components/LoginComponent.vue";
import { expect, it, describe, beforeEach } from "vitest";
import { Form, Field, ErrorMessage } from "vee-validate";

describe("LoginComponent", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(LoginComponent, {
      global: {
        components: {
          Form,
          Field,
          ErrorMessage,
        },
      },
    });
  });

  it("renders correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("shows an error message if email field is empty", async () => {
    const emailField = wrapper.find('input[type="email"]');
    const errorMessage = wrapper.find(".error-message");

    expect(errorMessage.text()).toBe("");
    await emailField.setValue("");
    await emailField.trigger("blur");

    expect(errorMessage.text()).toBe("This field is required");
  });

  it("shows an error message if password field is empty", async () => {
    const passwordField = wrapper.find('input[type="password"]');
    const errorMessage = wrapper.find(".error-message");

    expect(errorMessage.text()).toBe("");
    await passwordField.setValue("");
    await passwordField.trigger("blur");

    expect(errorMessage.text()).toBe("This field is required");
  });
});
