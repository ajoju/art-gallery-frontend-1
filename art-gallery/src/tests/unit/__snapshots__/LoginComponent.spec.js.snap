// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginComponent > renders correctly 1`] = `
"<div data-v-87276d9b=\\"\\" class=\\"container\\">
  <div data-v-87276d9b=\\"\\" class=\\"card\\">
    <form data-v-87276d9b=\\"\\" novalidate=\\"\\" name=\\"login-form\\">
      <p data-v-87276d9b=\\"\\" class=\\"login-text\\">Enter your login details to sign in, or click 'Sign up' to create an account.</p><label data-v-87276d9b=\\"\\" for=\\"email\\">Email</label>
      <p data-v-87276d9b=\\"\\" class=\\"error-message\\">
        <!---->
      </p><input data-v-87276d9b=\\"\\" type=\\"email\\" placeholder=\\"Email\\" class=\\"form-control\\" name=\\"email\\"><label data-v-87276d9b=\\"\\" for=\\"password\\">Password:</label>
      <p data-v-87276d9b=\\"\\" class=\\"error-message\\">
        <!---->
      </p><input data-v-87276d9b=\\"\\" type=\\"password\\" placeholder=\\"Password\\" class=\\"form-control\\" name=\\"password\\">
      <div data-v-87276d9b=\\"\\" class=\\"button-div\\"><button data-v-87276d9b=\\"\\" type=\\"submit\\" class=\\"login-submit\\">Log In</button><button data-v-87276d9b=\\"\\">
          <router-link data-v-87276d9b=\\"\\" to=\\"/signup\\" class=\\"btn-link\\">Sign up</router-link>
        </button></div>
    </form>
  </div>
</div>"
`;
